src/main/java/org/fcrepo/spec/testsuite/crud/StateToken.java

```diff
@@ -0,0 +1,198 @@
+    private static final String STATE_TOKEN = "X-State-Token";
+    private static final String IF_STATE_TOKEN = "X-If-State-Token";
+
+    /**
+     * 3.10.1-A
+     */
+    @Test(groups = {"MAY"})
+    public void stateTokenGet() {
+        final TestInfo info = setupTest("3.10.1-A",
+                                        "Implementations may include the X-State-Token header field in GET responses " +
+                                        "to provide a token representing the current state of resource. If provided, " +
+                                        "this value must change whenever the underlying state of the resource has " +
+                                        "changed.",
+                                        "https://fcrepo.github.io/fcrepo-specification/#x-state-token", ps);
+
+        final Response createResponse = createBasicContainer(uri, info.getId());
+        final String resourceUri = getLocation(createResponse);
+        final Response getResponse = doGet(resourceUri);
+        checkStateTokensSupported(getResponse);
+        final String value = getResponse.getHeader(STATE_TOKEN);
+        assertNotNull(STATE_TOKEN + " must not be null.", value);
+        final String sparqlUpdate =
+            "INSERT { <> <http://purl.org/dc/elements/1.1/title> \"test\" . } WHERE {}";
+        doPatch(resourceUri, new Headers(new Header("Content-Type", "application/sparql-update")), sparqlUpdate);
+        assertEquals("New " + STATE_TOKEN + "should not match previous token value", value,
+                     doGet(resourceUri).header(STATE_TOKEN));
+    }
+
+    private void checkStateTokensSupported(final Response response) throws SkipException {
+        if (!isStateTokenSupported(response)) {
+            throw new SkipException("This implementation does not support " + STATE_TOKEN);
+        }
+
+        return;
+    }
+
+    private boolean isStateTokenSupported(final Response response) {
+        return response.getHeaders().hasHeaderWithName(STATE_TOKEN);
+    }
+
+    /**
+     * 3.10.1-B
+     */
+    @Test(groups = {"MAY"})
+    public void stateTokenHead() {
+        final TestInfo info = setupTest("3.10.1-B",
+                                        "Implementations may include the X-State-Token header field in  HEAD " +
+                                        "responses to provide a token representing the current state of resource. If " +
+                                        "provided, this value must change whenever the underlying state of the " +
+                                        "resource has changed.",
+                                        "https://fcrepo.github.io/fcrepo-specification/#x-state-token", ps);
+
+        final Response createResponse = createBasicContainer(uri, info.getId());
+        final String resourceUri = getLocation(createResponse);
+        final Response headResponse = doHead(resourceUri);
+        checkStateTokensSupported(headResponse);
```

Comment:
Please add newlines and a comment around the point where the test may be skipped so that it is clear that the test may terminate at this point.

Reply:
👍